elf2flt.c

```diff
@@ -1948,7 +1951,14 @@ int main(int argc, char *argv[])
 	  | (pic_with_got ? FLAT_FLAG_GOTPIC : 0)
 	  | (docompress ? (docompress == 2 ? FLAT_FLAG_GZDATA : FLAT_FLAG_GZIP) : 0)
 	  );
-  hdr.build_date = htonl((uint32_t)time(NULL));
+
+  sde = getenv("SOURCE_DATE_EPOCH");
+  if (sde)
+      t = (time_t) strtoul(sde, NULL, 10);
```

Comment:
The toolchain wrapper in Buildroot use `strtoll()` and checks for `errno` after the attempted conversion.

Reply:
And as usual, reading the [docs](https://reproducible-builds.org/docs/source-date-epoch/) is insightful:
they use `strtoull()`.